src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/DataContractJsonSerializer.cs

```diff
@@ -615,6 +625,9 @@ private void Initialize(Type type,
             _serializeReadOnlyTypes = serializeReadOnlyTypes;
             _dateTimeFormat = dateTimeFormat;
             _useSimpleDictionaryFormat = useSimpleDictionaryFormat;
+
+            // Reset _rootContract when parameters change so it gets recalculated with surrogate
+            _rootContract = null;
```

Comment:
This "cache invalidation" is not necessary and it is not done in the big brother DataContractSerializer. Please undo this change.